store/store.go

```diff
@@ -270,8 +271,8 @@ func respToError(resp *http.Response, msg string) error {
 	return fmt.Errorf(tpl, msg, resp.StatusCode, resp.Request.Method, resp.Request.URL)
 }
 
-func getStructFields(s interface{}) []string {
-	st := reflect.TypeOf(s)
+func getStructFields(s interface{}, exceptions ...string) []string {
```

Comment:
API wise, I wonder if we should have a `getStructFields` and `getStructFieldsWithExceptions` ? Looking at the usage of `getStructFields` its not clear what the second arg means (without looking at the definition).